examples/vision/image_captioning.py

```diff
@@ -0,0 +1,486 @@
+"""
+
+
+"""shell
+! wget -q https://github.com/jbrownlee/Datasets/releases/download/Flickr8k/Flickr8k_Dataset.zip
+! wget -q https://github.com/jbrownlee/Datasets/releases/download/Flickr8k/Flickr8k_text.zip
+! unzip -qq Flickr8k_Dataset.zip
+! unzip -qq Flickr8k_text.zip
+! rm Flickr8k_Dataset.zip Flickr8k_text.zip
+"""
+
+
+# Path to the images
+IMAGES_PATH = "Flicker8k_Dataset"
+
+# Desired image dimensions
+IMAGE_SIZE = (299, 299)
+
+# Vocabulary size
+VOCAB_SIZE = 10000
+# Maximum length allowed for any sequence
+MAX_SEQ_LENGTH = 40
+
+# Dimensions for the image embeddings and token embeddings
+EMBED_DIM = 512
+# Number of self-attention heads
+NUM_HEADS = 2
+# Dimensions of the feed-forward network
+FF_DIM = 512
+
+# Other training parameters
+BATCH_SIZE = 32
+EPOCHS = 15
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+
+
+"""
+## Preparing the dataset
+"""
+
+
+def load_captions_data(filename):
+    """Load captions (text) data and maps them to corresponding images.
+
+    Args:
+        filename: path to the text file containing caption data.
+
+    Returns:
+        caption_mapping: dictionary mapping image names and the corresponding captions
+        text_data: list containing all the available captions
+    """
+
+    with open(filename) as caption_file:
+        caption_data = caption_file.readlines()
+        caption_mapping = {}
+        text_data = []
+
+        for line in caption_data:
+            line = line.rstrip("\n")
+            # Image name and captions are separated using a tab
+            img_name, caption = line.split("\t")
+            # Each image is repeated five times for the five different captions. Each
+            # image name has a prefix `#(caption_number)`
+            img_name = img_name.split("#")[0]
+            img_name = os.path.join(IMAGES_PATH, img_name.strip())
+
+            if img_name.endswith("jpg"):
+                # We will add a start and an end token to each caption
+                caption = "<start> " + caption.strip() + " <end>"
+                text_data.append(caption)
+
+                if img_name in caption_mapping:
+                    caption_mapping[img_name].append(caption)
+                else:
+                    caption_mapping[img_name] = [caption]
+
+        return caption_mapping, text_data
+
+
+def train_val_split(caption_data, train_size=0.8, shuffle=True):
+    """Split the captioning dataset into train and validation sets.
+
+    Args:
+        caption_data (dict): a dictionary containing the mapped caption data
+        train_size (float): fraction of all the full dataset to use as training data
+        shuffle (bool): Whether to shuffle the dataset before splitting
+
+    Returns:
+        Traning and validation datasets as two separated dicts
+    """
+
+    # 1. Get the list of all image names
+    all_images = list(caption_data.keys())
+
+    # 2. Shuffle if necessary
+    if shuffle:
+        np.random.shuffle(all_images)
```

Comment:
I recommend seeding. You might get wildly different results from run to run if you train on different splits every time.

Reply:
I was using seed in my original experiment, it might have been removed accidentally while converting the notebook to py file and fromatting

Reply:
Please add it back.

Reply:
Added the seed for both `numpy` as well as `tf`